### ISSUE TITLE
Modifier keys?

### ISSUE THREAD

**Issue by minorninth** [chaals/accesskey#16]
Here are some examples of real-world shortcuts implemented by existing web apps:
- 'C' to compose a new message in Gmail/Inbox
- Ctrl+Shift+C to do a Word Count in Google Docs
- Shift+A to "reply all" in Yahoo Mail
- 'L' to like the current story on Facebook
- '/' to focus the search box on Twitter
- 'C' to create an issue on GitHub

It doesn't sound like any of these apps could switch to accesskey, because accesskey only supports the browser modifier keys.

Is there any reason accesskey couldn't be expanded to support key combinations with modifier keys explicitly, like Shift+A or Ctrl+Shift+C?

**Comment by chaals** [comment]
The only barrier here is what people are willing to implement in browsers.

The spec could, and IMHO **should**, support requesting specific modifier keys - or bare keys, which are obviously more convenient for most users most of the time.

The first step would be to fix the bug in most browsers that means if there is more than one character as the value of the attribute, then it is not recognised at all.

The second step, which I think is pretty trivial would be to define a microsyntax for such modifiers.

Similarly, I would be happier if `accessKeyLabel` returned a defined microsyntax that could be parsed, rather than the HTML5 situation of "user agents can describe the accesskey label however they like, so it cannot be parsed reliably".

**Comment by minorninth** [comment]
I defined such a microsyntax in the aria-keyshortcuts spec, maybe consider borrowing some of that text?

**Comment by chaals** [comment]
I was planning on copying the DOM specs for the most part, so yes…

**Comment by chaals** [comment]
Yep. I'm going to say that browsers must understand - and use - the DOM UI Events syntax for describing keys.

**Comment by Krinkle** [comment]
@chaals Sorry if this is not the right, place. I repeatedly see this desire for the label to be parseable. I don't oppose that, but I'm curious why this is so important, and specifically why it is important for this property to be parseable (as opposed to a new getter optimised for machine-readability as opposed to human-readability). Is there a particular use case you have in mind?

**Comment by chaals** [comment]
@Krinkle I am quite happy for it to optimise for machine-readability - that's the critical requirement, since then it is possible to present to the user in a localised and appropriate form.

The sceneario I have in mind is the application presenting the shortcut to the user, e.g. in help documentation or some configuration system. The point is that this should work effectively, rather than having the author trying to cover all the possible cases, or doing something fragile (and not really good) like user-agent sniffing and then guessing.

**Comment by Malvoz** [comment]
Not trying to ruin all the fun here, but multiple modifier keys sounds pretty _inaccessible_ to me.
